File: 1.swift
var depths: [Int] = []

while let line = readLine() {
    depths.append(Int(line)!)
}

var total = 0

for i in 1..<depths.count {
    if depths[i - 1] < depths[i] {
        total += 1
    }
}

print("a:", total)

var prev = 1000000
total = 0

for i in 1..<depths.count - 2 {
    let new = depths[i..<i + 3].reduce(0, +)

    if new > prev {
        total += 1
    }
    prev = new
}
print("b:", total)

File: 01/1.py
import sys

depths = []

for line in sys.stdin:
    depths.append(int(line))

total = 0
for i in range(1, len(depths)):
    if depths[i - 1] < depths[i]:
        total += 1

print("a:", total)


prev = 1000000
total = 0
for i in range(len(depths) - 2):
    new = sum(depths[i : i + 3])

    if new > prev:
        total += 1
    prev = new

print("b:", total)
